### PR TITLE
Google fonts

### PR DESCRIPTION
**Pull request by alexeymorozov**
PR for #108 

The script imports 1017 of 1111 fonts, leaving 94 skipped fonts (see #122).

Font is skipped if there is no metadata file or a download links return 404.

Some fonts could have empty license text.

Specs are coming.

### CODE REVIEW

**Comment by ronaldtse** [comment]
Looks good @alexeymorozov !

There are two failures in specs right now:

```
  1) Fontist::Formulas::MontserratFont initializing builds the data dictionary
197
     Failure/Error: expect(formula.fonts.count).to eq(2)
198

199
       expected: 2
200
            got: 3
201

202
       (compared using ==)
203
     # ./spec/fontist/formulas/montserrat_font_spec.rb:8:in `block (3 levels) in <top (required)>'
204

205
  2) Fontist::Formulas::OverpassFont initializing builds the data dictionary
206
     Failure/Error: expect(formula.fonts.count).to eq(2)
207

208
       expected: 2
209
            got: 3
210

211
       (compared using ==)
212
     # ./spec/fontist/formulas/overpass_font_spec.rb:8:in `block (3 levels) in <top (required)>'
213

214
```

Could you please fix it? Thanks.

**Comment by alexeymorozov** [comment]
Yeap. Now we have two copies of Montserrat and Overpass fonts, which seem identical.

Could keep a Google's variant or a current one. The Google's could be autoupdated in future.

**Comment by ronaldtse** [comment]
@alexeymorozov - aha -- the previous Montserrat and Overpass fonts were downloaded from the source instead of Google. In addition, there are some other formulas like Lato (#117), where Google is serving Lato 1.0, but the Lato homepage serves 2.0. 

Maybe we should have a "google override" so that the Google import script skips Monserrat, Overpass and Lato?

**Comment by alexeymorozov** [comment]
I think it's a good solution. Will do it

**Comment by ronaldtse** [comment]
Looks good to me, if there are no further changes we can merge this.

File: lib/fontist/formulas/google/zcool_qingke_huangyou_font.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        THE FONT SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+        EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO ANY WARRANTIES OF
+        MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
+        OF COPYRIGHT, PATENT, TRADEMARK, OR OTHER RIGHT. IN NO EVENT SHALL THE
+        COPYRIGHT HOLDER BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+        INCLUDING ANY GENERAL, SPECIAL, INDIRECT, INCIDENTAL, OR CONSEQUENTIAL
+        DAMAGES, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+        FROM, OUT OF THE USE OR INABILITY TO USE THE FONT SOFTWARE OR FROM
+        OTHER DEALINGS IN THE FONT SOFTWARE.
+      EOS
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/fontist/formulas/google/zcool_qingke_huangyou_font.rb:142]: @alexeymorozov see if you can fix this?

File: lib/fontist/import/google_import.rb [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        license = File.read(file).chomp.gsub("\r\n", "\n") if file
+        { license: license || '' }
+      end
+
+      def from_otfinfo(path)
+        font_file = Dir.glob(File.join(path, '*.ttf')).first
+        text = `otfinfo --info #{font_file}`
+        otf = text.split("\n")
+                  .select { |x| x.include?(':') }
+                  .map    { |x| x.split(':', 2).map(&:strip) }
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/fontist/import/google_import.rb:141]: @alexeymorozov see if you can fix this?

File: lib/fontist/import/google_import.rb [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        print "warn, no license, " unless file
+        license = File.read(file).chomp.gsub("\r\n", "\n") if file
+        { license: license || '' }
+      end
+
+      def from_otfinfo(path)
+        font_file = Dir.glob(File.join(path, '*.ttf')).first
+        text = `otfinfo --info #{font_file}`
+        otf = text.split("\n")
+                  .select { |x| x.include?(':') }
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/fontist/import/google_import.rb:140]: @alexeymorozov see if you can fix this?

File: lib/fontist/import/google_import.rb [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+          style = style(f, taken)
+          taken << style
+          styles << OpenStruct.new(style: style, filename: f.filename)
+        end
+
+        styles
+      end
+
+      def style(font, taken)
+        match_style(font.filename) || default_if_first(taken) || match_style(font.post_script_name) || default_style
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/fontist/import/google_import.rb:113]: @alexeymorozov see if you can split this line

File: lib/fontist/import/google_import.rb [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+        # Protobuf file could be downloaded from
+        # https://raw.githubusercontent.com/googlefonts/gftools/master/Lib/gftools/fonts_public.proto
+        #
+        # To compile Protobuf to Ruby use `ruby-protoc lib/fontist/import/google/fonts_public.proto`
+        Google::Fonts::FamilyProto.parse_from_text(File.read(metadata_path))
+      end
+
+      def in_skiplist?(name)
+        @skiplist ||= YAML.load(File.open(File.expand_path("../google/skiplist.yml", __FILE__)))
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/fontist/import/google_import.rb:58]: @alexeymorozov see if you can fix this?

File: lib/fontist/import/google/fonts_public.pb.rb [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      optional :float, :max_value, 4
+    end
+
+    class SourceProto < ::ProtocolBuffers::Message
+      set_fully_qualified_name "google.fonts.SourceProto"
+
+      optional :string, :repository_url, 1
+      optional :string, :commit, 2
+    end
+
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/fontist/import/google/fonts_public.pb.rb:69]: @alexeymorozov see if you can fix this?

**Comment by ronaldtse** [comment]
@alexeymorozov just some minor formatting issues as described above.

**Comment by alexeymorozov** [comment]
Thank you :) Yes, I'll fix them.

Will continue tomorrow.

**Comment by alexeymorozov** [comment]
Extracted fonts to another PR #123, so generated code is separated from the logic.

@ronaldtse Requesting for feedback, and could merge this one

**Comment by ronaldtse** [comment]
Thank you @alexeymorozov !

**Comment by alexeymorozov** [comment]
Thank you @ronaldtse :-)

**Pull request merged**